cordova-lib/src/cordova/platform.js

```diff
@@ -110,15 +110,25 @@ function addHelper(cmd, hooksRunner, projectRoot, targets, opts) {
                 if(platform === 'wp8') {
                     events.emit('warn', 'wp8 has been deprecated. Please use windows instead.');
                 }
-                if(spec && pkgJson) {
-                    pkgJson.dependencies[pkgJsonSpec] = spec;
+                // If there is a spec specified during add, add to pkg.json dependencies(if exists).
```

Comment:
Use case would be : if pkg.json has a spec (4.2.1), config has a spec (4.2.2), but the user passes in a specific version (let's say 4.0.0).  If pkg.json never gets written to, but config.xml does... during restore it will try to make them match by looking at pkg.json first, but that version would not have been updated and would still be 4.2.1. Does this use case make sense?